Dfc.CourseDirectory.Core/ReferenceData/Lars/LarsDataImporter.cs

```diff
@@ -23,185 +28,270 @@ public class LarsDataImporter
         private readonly HttpClient _httpClient;
         private readonly ISqlQueryDispatcherFactory _sqlQueryDispatcherFactory;
         private readonly ILogger<LarsDataImporter> _logger;
+        private readonly BlobContainerClient _blobContainerClient;
 
         public LarsDataImporter(
             HttpClient httpClient,
             ISqlQueryDispatcherFactory sqlQueryDispatcherFactory,
             ILogger<LarsDataImporter> logger,
-            IOptions<LarsDataset> larsDatasetOption)
+            IOptions<LarsDataset> larsDatasetOption, BlobServiceClient blobServiceClient)
         {
             _httpClient = httpClient;
             _sqlQueryDispatcherFactory = sqlQueryDispatcherFactory;
             _logger = logger;
             _larsDataset = larsDatasetOption.Value;
+            _blobContainerClient = blobServiceClient.GetBlobContainerClient(Constants.LarsContainerName);
         }
-
         public async Task ImportData()
         {
-            var extractDirectory = Path.Join(Path.GetTempPath(), "lars");
-            Directory.CreateDirectory(extractDirectory);
+            _logger.LogTrace("LarsDataImport started at: {time}", DateTimeOffset.Now);
+            try
+            {
+                var downloadDate = string.Empty;
+                //Read from blob storage
+                var blobClient = _blobContainerClient.GetBlobClient("larsdownloadinfo.json");
+                if (await blobClient.ExistsAsync())
+                {
+                    var downloadInfoContent = await blobClient.DownloadContentAsync();
+                    var downloadInfoJson = downloadInfoContent.Value.Content.ToString();
+                    var downloadInfo = JsonSerializer.Deserialize<Dictionary<string, string>>(downloadInfoJson);
+                    downloadDate = downloadInfo["LastDownloadDate"];
+                    _logger.LogTrace("Lars last downloaded on {downloadDate} ", downloadDate);
+                }
+                else
+                {
+                    downloadDate = "01/01/1900";
+                }
+                var lastDownloadDay = DateOnly.Parse(downloadDate);
```

Comment:
Nothing wrong with what is done here, however, I feel that the below is a little more explicit.

```suggestion
                var lastDownloadDay = DateOnly.TryParse(downloadDate, out var res) ? res : new DateOnly(1900, 1, 1);
```